asdf_astropy/converters/wcs/tests/test_wcs.py

```diff
@@ -0,0 +1,78 @@
+import numpy as np
+import pytest
+from astropy.wcs import WCS, DistortionLookupTable, Sip
+
+from asdf_astropy.testing.helpers import assert_wcs_roundtrip
+
+
+def create_empty_wcs():
+    return WCS()
+
+
+def create_wcs_with_attrs():
+    wcs = WCS(naxis=3)
+    wcs.pixel_shape = [100, 200, 300]
+    wcs.pixel_bounds = [[11, 22], [33, 45], [55, 67]]
+    return wcs
+
+
+def create_sip_distortion_wcs():
+    rng = np.random.default_rng(42)
+    wcs = WCS(naxis=2)
+    wcs.wcs.crval = [251.29, 57.58]
+    wcs.wcs.cdelt = [1, 1]
+    wcs.wcs.crpix = [507, 507]
+    wcs.wcs.pc = np.array([[7.7e-6, 3.3e-5], [3.7e-5, -6.8e-6]])
+    wcs._naxis = [1014, 1014]
+    wcs.wcs.ctype = ["RA---TAN-SIP", "DEC--TAN-SIP"]
+
+    # Generate random SIP coefficients
+    a = rng.uniform(low=-1e-5, high=1e-5, size=(5, 5))
+    b = rng.uniform(low=-1e-5, high=1e-5, size=(5, 5))
+
+    # Assign SIP coefficients
+    wcs.sip = Sip(a, b, None, None, wcs.wcs.crpix)
+    wcs.wcs.set()
+
+    return wcs
+
+
+def create_tabular_wcs():
```

Comment:
This is a distortion table, but do we have a test for a `-TAB` wcs?

Reply:
Any suggestions for an example to try?

If I use the one here:
https://github.com/astropy/astropy/blob/0f80be718107ef797a58ba42025e70b21c454c61/astropy/wcs/tests/conftest.py#L11
it doesn't roundtrip through pickle (or `to_fits`)
```python
import pickle
from astropy.wcs import WCS
from astropy.wcs.tests.helper import SimModelTAB
st = SimModelTAB(nx=150, ny=200)
hdulist = st.hdulist
wcs = WCS(hdulist[0].header, hdulist)
pickle.loads(pickle.dumps(wcs))
```
```
KeyError: "Extension ('WCS-TABLE', 1) not found."
```

Reply:
The file in the astropy package data `get_pkg_data_filename("data/tab-time-last-axis.fits", package="astropy.wcs.tests")` also fails to round-trip through pickle (or `to_fits`).

Reply:
Oh https://github.com/astropy/astropy/issues/9998 astropy can't produce a `-TAB` wcs so we can't support it here.

Reply:
Well I guess we should write a test which errors then :laughing: